sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatAsyncClient.java

```diff
@@ -44,16 +47,17 @@
     private final SignalingClient signalingClient;
     private final ChatImpl chatClient;
 
-    ChatAsyncClient(AzureCommunicationChatServiceImpl chatServiceClient, SignalingClient signalingClient) {
+    ChatAsyncClient(AzureCommunicationChatServiceImpl chatServiceClient) {
         this.chatServiceClient = chatServiceClient;
-        this.signalingClient = signalingClient;
+        this.signalingClient = new CommunicationSignalingClient();
```

Comment:
If I understand correctly - with the `new CommunicationSignalingClient()`, we ALWAYS loads the `TrouterClient` into JVM, earlier this was done only if the user-provided the notification params in the builder. I think the new aggressive instantiation will be a problem if the user wants to use "chat SDK without notification" + "calling SDK" in the same app, given loading two trouter is an issue at least now (that router team is working on).

Reply:
You're right. Maybe we could update to create signalingClient instance when user calls startRealtimeNotification the first time?

Reply:
Verified. With current implementation "the initialization of trouter is moved to start() method, so types should be loading only when start() is called", if I put chat & calling in the same App with excluding trouter from chat, the App could start and chat functionality works fine if you don't call startRealtimeNotifications().
We are good here.